### DISCUSSION TITLE
ComboBox - different templates for a selected item and an item in a popup

### DISCUSSION THREAD

**Discussion by blahetal** [AvaloniaUI/Avalonia#10287] [Q&A] [answered]
Hi, the code below sets a red background to a selected item in a ComboBox. When the ComboBox is open, I want the items in the popup to have a yellow background. What I need is a template for selected item (this one will be simple) and another one for items in the popup (here I plan to put more complex layout).
In wpf, I did it with help of `DataTemplateSelector` but I think n Avalonia it is not need it as I can access `ItemTemplate` ([link](https://github.com/AvaloniaUI/Avalonia/blob/release/0.10.18/src/Avalonia.Themes.Fluent/Controls/ComboBox.xaml#L142)) of the `ItemsPresenter` but it does not work - all items in the popup are red. I suspect that the style is not applied at all or I need to set it in some other way.

```
<ComboBox Items="{Binding X}" SelectedIndex="0">
	<ComboBox.Styles>
		<Style Selector="ComboBox /template/ ItemsPresenter#PART_ItemsPresenter">
			<Setter Property="ItemTemplate">
				<DataTemplate>
					<TextBlock Background="Yellow" Text="{Binding}" />
				</DataTemplate>
			</Setter>
		</Style>
	</ComboBox.Styles>

	<ComboBox.ItemTemplate>
		<DataTemplate>
			<TextBlock Background="Red" Text="{Binding}" />
		</DataTemplate>
	</ComboBox.ItemTemplate>
</ComboBox>
```

**Comment by ahopper** [comment, marked as answer]
this works with the fluent combobox for colors, I'm not sure if you can easily set a different template for each part.
```
<ComboBox.Styles>
	<Style Selector="ComboBox TextBlock">
		<Setter Property="Background" Value="Red"/>
	</Style>

	<Style Selector="ComboBox /template/ ContentControl#ContentPresenter TextBlock">
		<Setter Property="Background" Value="Yellow"/>
	</Style>
</ComboBox.Styles>
```

**Reply by blahetal** [reply]
thanks, your example gave me an idea. I did not find out how to set different templates tho but I can simulate it with showing/hiding elements in the popup

```xaml
<ComboBox Items="{Binding X}" SelectedIndex="0">
	<ComboBox.Styles>
		<Style Selector="ComboBox /template/ ContentControl#ContentPresenter > StackPanel > TextBlock:nth-child(2)">
			<Setter Property="IsVisible" Value="False" />
		</Style>
	</ComboBox.Styles>

	<ComboBox.ItemTemplate>
		<DataTemplate>
			<StackPanel Orientation="Horizontal">
				<TextBlock Grid.Column="0" Text="{Binding}" />
				<TextBlock Grid.Column="1" Text="Details visible only in the popup" />
			</StackPanel>
		</DataTemplate>
	</ComboBox.ItemTemplate>
</ComboBox>
```

Changing `DataTemplate` would be ideal but I can live with this :)
